bin/midi.ml

```diff
@@ -50,6 +50,11 @@ let bend_pitch ~bend ~timestamp =
   let data2 = char_of_int (bend lsr 7) in
   Event.create ~status:'\224' ~data1 ~data2 ~timestamp
 
+let control_change ~cc ~value ~timestamp =
+  let data1 = char_of_int ((if cc > 119 then 119 else cc) land 0b1111111) in
```

Comment:
I would say here we can handle CC > 119 by just throwing an Invalid_arg exception (using `invalid_arg`.)
This is not pretty but is a common way to deal with arbitrary numerical restrictions on arguments.
You can document this behaviour in the `control_change` entry in the `.mli` file.

Reply:
Ok

Reply:
I was not very confident adding that condition because of these:

![Screenshot (581)](https://user-images.githubusercontent.com/65472478/229331378-f1182895-d0f1-4843-aec4-a5c30ff756ab.png)
[source](https://www.midi.org/specifications-old/item/table-1-summary-of-midi-message)

![Screenshot (582)](https://user-images.githubusercontent.com/65472478/229331393-ba2e5715-906d-4ca1-b30a-f87f97e0e0db.png)
[source](https://cmtext.indiana.edu/MIDI/chapter3_controller_change.php)

Reply:
IIRC, 120-127 is for _program change_, which is not quite use for the same purposes: it is fine to me to exclude things above 127, we'll adapt in the future if required. 😊

Reply:
Ok